### PR TITLE
Amp 9 healthcheck endpoint

### PR DESCRIPTION
**Pull request by wajerrr**
- added healthcheck endpoint
- split our routes into modules
- added test for amp-rendering handler
- fixed favicon server call causing 404 from GraphQL

### CODE REVIEW

File: src/server/template.js [hunk 1]
```diff
@@ -10,12 +10,15 @@ export const hotReloadingScript = `<script src="/socket.io/socket.io.js"></scrip
    window.location.reload();
   });
 </script>`;
+const iconBase64 =
+  ' iVBORw0KGgoAAAANSUhEUgAAABAAAAAQAQMAAAAlPW0iAAAABGdBTUEAALGPC/xhBQAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAABlBMVEXjEgv////4KLUSAAAAAWJLR0QB/wIt3gAAAAd0SU1FB+IFEAgALURtxQsAAAAMSURBVAjXY2AgDQAAADAAAceqhY4AAAAldEVYdGRhdGU6Y3JlYXRlADIwMTgtMDUtMTZUMDg6MDA6NDUtMDc6MDDueH9GAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDE4LTA1LTE2VDA4OjAwOjQ1LTA3OjAwnyXH+gAAAABJRU5ErkJggg==';
```
Comment (yasgreen93) [inline src/server/template.js]: What is this icon?

File: .ebextensions/01-heanthcheck.config [hunk 2]
```diff
@@ -0,0 +1,5 @@
+option_settings:
+  aws:elasticbeanstalk:environment:process:default:
+    HealthCheckPath: /healthcheck
+  aws:elasticbeanstalk:application:
+    Application Healthcheck URL: /healthcheck
```
Comment (yasgreen93) [inline .ebextensions/01-heanthcheck.config]: New line at the end of the file here?

File: src/server/handlers/amp-page-render.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import getGraphqlData from '../get-graphql-data';
+import renderHtml from '../render-html';
+
+export const generateRefUrl = (pathname) =>
+  `https://www.economist.com/${pathname ||
+    'news/world-week/21741222-politics-week'}`;
```
Comment (yasgreen93) [inline src/server/handlers/amp-page-render.js]: If this article is being re-used in multiple places shall we put it in a const and re-use it?

File: src/server/handlers/amp-page-render.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable no-console */
+import getGraphqlData from '../get-graphql-data';
+import renderHtml from '../render-html';
+import server from '../server';
+
+const mockData = 'testData';
+
+const location = 'https://stickyfloors.net';
```
Comment (yasgreen93) [inline src/server/handlers/amp-page-render.test.js]: What is `https://stickyfloors.net`?

File: src/server/handlers/amp-page-render.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import getGraphqlData from '../get-graphql-data';
+import renderHtml from '../render-html';
+
+export const generateRefUrl = (pathname) =>
+  `https://www.economist.com/${pathname ||
```
Comment (umbobabo) [inline src/server/handlers/amp-page-render.js]: We should move hardcoded value in a getter function thinking about AMP as a service. Given the requested page you can get from a mapping the correct domain.
Something like...
```
getDomain(pathname <<< 'This is amp.economist.com') should return 'www.economist.com'
```
Male sense?

File: src/server/handlers/healthcheck.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import fs from 'fs';
+import { execSync } from 'child_process';
+import config from '../config';
+
+let lastCommit = 'unknown';
+try {
+  // git will resolve outside a container, `/code/last_commit.txt` is pushed into a container
```
Comment (umbobabo) [inline src/server/handlers/healthcheck.js]: We have a standard for comments that is Capital letter at the begin of the string and full stop at the end.
```
// Git will resolve outside a container, `/code/last_commit.txt` is pushed into a container.
```
The comment is not very clear. Think about the fact that a junior can read this too.

File: src/server/template.js (hunk 1 above)
Comment (umbobabo) [reply src/server/template.js]: favico.

File: src/server/handlers/amp-page-render.js (hunk 3 above)
Comment (umbobabo) [reply src/server/handlers/amp-page-render.js]: That was a default value while developing. I guess we have just to get rid of it

File: src/server/handlers/amp-page-render.test.js (hunk 4 above)
Comment (umbobabo) [reply src/server/handlers/amp-page-render.test.js]: I guess just a mock url

File: src/server/template.js (hunk 1 above)
Comment (wajerrr) [reply src/server/template.js]: favicon (stops browser asking server for `favicon.ico` (witch generates GraphQL 404 error)

File: .ebextensions/01-heanthcheck.config (hunk 2 above)
Comment (wajerrr) [reply .ebextensions/01-heanthcheck.config]: done

File: src/server/handlers/healthcheck.js (hunk 6 above)
Comment (wajerrr) [reply src/server/handlers/healthcheck.js]: well this was copied from fe-blogs.
but I have changed it now so it checks last commit only in production. (as we don't need it at all in while developing)

File: src/server/handlers/amp-page-render.js (hunk 5 above)
Comment (wajerrr) [reply src/server/handlers/amp-page-render.js]: makes sense. made stub `getDomain` method

File: src/server/handlers/amp-page-render.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (wajerrr) [reply src/server/handlers/amp-page-render.js]: removed

File: src/server/handlers/amp-page-render.test.js (hunk 4 above)
Comment (wajerrr) [reply src/server/handlers/amp-page-render.test.js]: just mock url, changed name and url to be more explicit (btw stickyfloors.net, it very good website! (no shameless plug here))

File: src/server/handlers/amp-page-render.js (hunk 3 above)
Comment (wajerrr) [reply src/server/handlers/amp-page-render.js]: removed now, as this was just temporary

File: src/server/template.js (hunk 1 above)
Comment (yasgreen93) [reply src/server/template.js]: Ah ok great!

**Pull request merged**